File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/MessageHeader.ts
import { CodeableConcept } from "./CodeableConcept";
import { Coding } from "./Coding";
import { ContactPoint } from "./ContactPoint";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { BackboneElement } from "./BackboneElement";

export type MessageHeaderResponse = BackboneElement & {
    code: string;
    details?: Reference;
    identifier: string;
};

export type MessageHeaderSource = BackboneElement & {
    name?: string;
    contact?: ContactPoint;
    version?: string;
    endpoint: string;
    software?: string;
};

export type MessageHeaderDestination = BackboneElement & {
    name?: string;
    target?: Reference;
    endpoint: string;
    receiver?: Reference;
};

export type MessageHeader = DomainResource & {
    response?: MessageHeaderResponse;
    definition?: string;
    enterer?: Reference;
    source: MessageHeaderSource;
    author?: Reference;
    reason?: CodeableConcept;
    responsible?: Reference;
    event?: string | Coding;
    sender?: Reference;
    focus?: Reference[];
    destination?: MessageHeaderDestination[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MessageHeader.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Coding
from base import ContactPoint
from base import DomainResource
from base import Reference
from base import BackboneElement

@dataclass(kw_only=True)
class MessageHeaderResponse(BackboneElement):
    code: str
    details: Optional[Reference] = None
    identifier: str

@dataclass(kw_only=True)
class MessageHeaderSource(BackboneElement):
    name: Optional[str] = None
    contact: Optional[ContactPoint] = None
    version: Optional[str] = None
    endpoint: str
    software: Optional[str] = None

@dataclass(kw_only=True)
class MessageHeaderDestination(BackboneElement):
    name: Optional[str] = None
    target: Optional[Reference] = None
    endpoint: str
    receiver: Optional[Reference] = None

@dataclass(kw_only=True)
class MessageHeader(DomainResource):
    response: Optional[MessageHeaderResponse] = None
    definition: Optional[str] = None
    enterer: Optional[Reference] = None
    source: MessageHeaderSource
    author: Optional[Reference] = None
    reason: Optional[CodeableConcept] = None
    responsible: Optional[Reference] = None
    sender: Optional[Reference] = None
    focus: Optional[List[Reference]] = field(default_factory=list)
    event_uri: Optional[str] = None
    destination: Optional[List[MessageHeaderDestination]] = field(default_factory=list)
    event_coding: Optional[Coding] = None